### PR TITLE
Falco Incubation Proposal

### PR DESCRIPTION
**Pull request by mfdii**
Falco is currently a CNCF Sandbox Project. Falco will present its annual review on the [October 15th, 2019 CNCF TOC call](https://github.com/cncf/toc#meeting-time). As part of this review, we would like the Falco project to be considered for Incubation. 

Signed-off-by: Michael Ducy <michael@ducy.org>

### CODE REVIEW

**Comment by markyjackson-taulia** [comment]
+1 this

File: reviews/incubation-falco.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+| Downloads | 43,482 | 121,270 (+179%) | dpkg/rpm |
+| Integrations | 8 | 40 (+400%) | 8 CNCF Projects |
+
+\* Pre-Sandbox is a 29 month period (May 2016-Oct 2018)
+
+### Integrations
+
+Integrations have been a major focus of the Falco team over the last year. Falco's usefulness is magnified when its data can be easily sent to an end user's existing Ops and SRE tooling. The Falco project can count over 40 integrations in a number of categories.
+
+- **Orchestrators:** Kuberenetes, Mesosphere
```
Comment (raravena80) [inline reviews/incubation-falco.md]: Minor typo: Kubernetes

File: reviews/incubation-falco.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+## Project Progress since Sandbox
+
+In short, the Falco project has seen terrific growth and project participation since entering the CNCF Sandbox. 
+
+* All key metrics around contributions and committers have seen significant growth.
+* Downloads have seen significant growth.
+* The project has seen an increase in integrations, as well as Falco being incorporated into other products (Sumo Logic, Altran).
+* The community has seen increased participation and activity.
+* Public end users are documented in the projects' [ADOPTERS.md](https://github.com/falcosecurity/falco/blob/dev/ADOPTERS.md). Two end users are speaking at Kubecon NA 2019.
+* The project completed a security audit and fixed 14 security bugs.
```
Comment (caniszczyk) [inline reviews/incubation-falco.md]: is the security audit published? I can't find it easily

Comment (mfdii) [reply reviews/incubation-falco.md]: I should revise that. We've fixed 13/14 bugs and will publish the audit when the last one is fixed. We are aiming to get this in next couple of releases.

File: reviews/incubation-falco.md (hunk 1 above)
Comment (mfdii) [reply reviews/incubation-falco.md]: fixed, thanks.

File: reviews/incubation-falco.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
+# Falco Incubation Stage Review Proposal
+
+The goal of the Falco project is to provide cloud native runtime security with a focus on Kubernetes intrusion & anomaly detection. Falco achieves this through the use of eBPF, Linux kernel probes, and the Kubernetes audit logging API to create an event stream of low level system and orchestrator events. This includes network connection info, file opens/writes, process information, and Kubernetes API calls. Falco also polls metadata (labels, resource names, container info, etc) from the Kubernetes API server and the underlying container runtime. 
+
+The Falco engine applies rule sets (or policies) to this event stream to identify potentially malicious behavior, or to simply log activity on a Kubernetes cluster for later auditing or review. The Kubernetes metadata can be incorporated into the rule conditions to allow rules to apply to only particular resources in a Kubernetes cluster. Falco alerts can be sent to a wide range of destinations including logging aggregation services, pub/sub services, serverles functions, and more. This allows Falco users to integrate Falco alerts into their existing incident response and Ops/SRE workflows.
```
Comment (mike-stewart) [inline reviews/incubation-falco.md]: Typo "serverles functions"
```suggestion
The Falco engine applies rule sets (or policies) to this event stream to identify potentially malicious behavior, or to simply log activity on a Kubernetes cluster for later auditing or review. The Kubernetes metadata can be incorporated into the rule conditions to allow rules to apply to only particular resources in a Kubernetes cluster. Falco alerts can be sent to a wide range of destinations including logging aggregation services, pub/sub services, serverless functions, and more. This allows Falco users to integrate Falco alerts into their existing incident response and Ops/SRE workflows.
```

**Comment by rajibmitra** [comment]
+1

**Comment by madhuakula** [comment]
+1 for this awesome project :+1:

**Comment by ricardolupo** [comment]
👍 👍 👍

**Comment by caniszczyk** [comment]
@jbeda has committed to do the due diligence on behalf of the TOC

**Comment by ariguillegp** [comment]
+1

**Comment by DannyPat44** [comment]
+1

**Comment by radhikapc** [comment]
+1

**Comment by recollir** [comment]
+1

**Comment by faevourite** [comment]
+1

**Comment by farshad-hobsons** [comment]
+1

**Comment by ricardolupo** [comment]
👍 👍 👍

**Comment by thecrudge** [comment]
+1

**Comment by billyshambrook** [comment]
We have been extremely happy with this tool. We are running over a 100 EC2 instances, and 100,000s of containers per day, and Falco handles the load fine.

On top of that, the support that we have had from the Falco contributers has been one of the best. 

A big +1 from us!!!!

**Comment by Kaizhe** [comment]
+1

**Comment by daviddetorres** [comment]
+1

**Comment by mfdii** [comment]
One question I had come up regarding Falco and our incubation proposal is adoption and end users. While we are just about done with the voting process, I'd still like to share this video from Kubecon NA 2020.

Kris Nova co presented with Abhinav Srivastava of Frame.io on the event processing pipeline they've built for Falco. In my opinion this is one of the better examples of Falco use cases:

Abhinav's presentation starts at 45:44 (and start at 0:00 if you want a great explanation of Falco from Kris) - 
https://youtu.be/Z4POV5IXnHQ?t=2744 

We're hoping to have a Frame.io case study published in the coming months.

One of the Falco project's goals as we move into Incubation is to increase the number and quality of end users. Currently our end users tend to come from tech, retail, or healthcare. We want to expand this out to include more financial services, fintech, and other regulated industries. We feel there's a opportunity to help these verticals with compliance use cases, as well as expand our current compliance use cases for retail and healthcare.

Thank you to the TOC (and the community) for all your support thus far!

**Comment by caniszczyk** [comment]
+1 binding votes (7/9)
Matt: https://lists.cncf.io/g/cncf-toc/message/3922
Michelle: https://lists.cncf.io/g/cncf-toc/message/3931
Joe: https://lists.cncf.io/g/cncf-toc/message/3952
Jeff: https://lists.cncf.io/g/cncf-toc/message/3963
Xiang: https://lists.cncf.io/g/cncf-toc/message/3967
Liz: https://lists.cncf.io/g/cncf-toc/message/3968
Alexis: https://lists.cncf.io/g/cncf-toc/message/3970

**Pull request merged**